Fixes use of <linux/usb.h> from userspace.

Update 2005-07-22 gotom, usb_ch9.h part is removed because it's
already applied to upstream.  New usb_ch9.h part is introduced
because the latest header has another broken usage.

Update 2005-05-05 gotom, usb_ch9.h should include linux/types.h
instead of asm/types.h due to install __le16.

--- include.orig/linux/usb.h	2005-04-21 20:04:32.000000000 +0900
+++ include/linux/usb.h	2005-05-05 17:20:59.610762324 +0900
@@ -1,7 +1,6 @@
 #ifndef __LINUX_USB_H
 #define __LINUX_USB_H
 
-#include <linux/mod_devicetable.h>
 #include <linux/usb_ch9.h>
 
 #define USB_MAJOR			180
@@ -19,6 +18,7 @@
 #include <linux/fs.h>		/* for struct file_operations */
 #include <linux/completion.h>	/* for struct completion */
 #include <linux/sched.h>	/* for current && schedule_timeout */
+#include <linux/mod_devicetable.h>
 
 struct usb_device;
 struct usb_driver;
--- include/linux/usb_ch9.h	2005-07-13 13:46:46.000000000 +0900
+++ include/linux/usb_ch9.h.gotom	2005-07-22 15:35:20.694654077 +0900
@@ -401,7 +401,6 @@
 
 	__u8  tTKID[3];
 	__u8  bReserved;
-	__u8  bKeyData[0];
 };
 
 /*-------------------------------------------------------------------------*/
@@ -526,7 +525,7 @@
 	USB_SPEED_UNKNOWN = 0,			/* enumerating */
 	USB_SPEED_LOW, USB_SPEED_FULL,		/* usb 1.1 */
 	USB_SPEED_HIGH,				/* usb 2.0 */
-	USB_SPEED_VARIABLE,			/* wireless (usb 2.5) */
+	USB_SPEED_VARIABLE			/* wireless (usb 2.5) */
 };
 
 enum usb_device_state {
